clients/rest-client-core/src/main/java/uk/gov/justice/services/clients/core/RestClientProcessor.java

```diff
@@ -163,9 +163,17 @@ private WebTarget createWebTarget(final EndpointDefinition definition, final Jso
             if (!payload.containsKey(paramName) && queryParam.isRequired()) {
                 throw new IllegalStateException(format("Query parameter %s is required, but not present in envelope", paramName));
             }
-
             if (payload.containsKey(paramName)) {
-                target = target.queryParam(paramName, payload.getString(paramName));
+                switch (queryParam.getType()) {
+                    case NUMERIC:
+                        target = target.queryParam(paramName, payload.getInt(paramName));
```

Comment:
Is getInt the correct name for this, assuming this covers both decimal and non-decimal values?